crates/bevy_asset/src/handle.rs

```diff
@@ -373,34 +380,32 @@ impl HandleUntyped {
     /// If this handle is strong and dropped, there is no guarantee that the asset
     /// will still be available (if only the returned handle is kept)
     pub fn typed_weak<T: Asset>(&self) -> Handle<T> {
-        self.clone_weak().typed()
+        self.clone_weak().typed().unwrap()
     }
 
     /// Converts this handle into a typed [`Handle`] of an [`Asset`] `T`.
     ///
     /// The new handle will maintain the Strong or Weak status of the current handle.
```

Comment:
```suggestion
    /// The new handle will maintain the Strong or Weak status of the current handle.
    /// If the handle is not of type `T`, a [`HandleError`] will be emitted returned.
```

Reply:
I didn't actually change the return type of this function, should we modify it as well?